03_how_ln_works.asciidoc

```diff
@@ -89,518 +93,526 @@ Also, in case you are not familiar with P2PKH addresses and the basic format and
-3. The Lightning Network is a dynamic organism. It changes constantly and frequently. Nodes are being added, other nodes are being turned off, balances change, etc. Even if everything is always communicated, the information will be valid only for a short amount of time. As a matter of fact, information might be already outdated by the time it is received. 
+
+1. To protect the privacy of the users and not shout out every financial transaction and payment that is being conducted. Channel balance updates would reveal that a payment has moved across the channel. This information could be correlated to reveal all payment sources and destinations.
+
+2. To scale the amount of payments that can be conducted with the Lightning Network. Remember that the Lightning Network was created in the first place because notifying every participant about every payment does not scale well. Thus, the Lightning Network cannot be designed in a way that balance updates of channels are  shared among participants.
+
+3. The Lightning Network is a dynamic system. It changes constantly and frequently. Nodes are being added, other nodes are being turned off, balances change, etc. Even if everything is always communicated, the information will be valid only for a short amount of time. As a matter of fact, information is often outdated by the time it is received.
+
+We will examine the details of the gossip protocol in a later chapter.
+
+For now, it is only important to know that the gossip protocol exists and that it is used to share topology information of the Lightning Network.
+This topology information is crucial for delivering payments through the network of payment channels.
+
+
+==== Path finding and routing
+
+Payments on the Lightning Network are forwarded along a _path_ made of channels linking one participant to another, from the payment source to the payment destination. The process of finding a path from source to destination is called _path finding_. The process of using that path to make the payment is called _routing_.
+
+[NOTE]
 ====
+A frequent criticism of the Lightning network is that "routing" is not solved, or even is an "unsolvable" problem. In fact, routing is trivial. Path finding, on the other hand, is a difficult problem. The two terms are often confused and need to be clearly defined to identify which problem we are attempting to solve.
+====
+
+As we will see next, the Lightning Network currently uses a _source-based onion-routed_ protocol for path finding and routing payments. Source-based means that the sender of the payment has to find a path through the network to the intended destination. Onion-routed means that the elements of the path are layered (like an onion), with each layer encrypted so that it can only be seen by one node at a time. We will discuss onion routing in the next section.
```

Comment:
not sure if it is accurate to say that it uses onion routing for path finding. I mean technically the probing does but it sounds a little weired.